llvm/lib/MC/DXContainerPSVInfo.cpp

```diff
@@ -168,7 +168,7 @@ void PSVRuntimeInfo::finalize(Triple::EnvironmentType Stage) {
   DXConStrTabBuilder.add(EntryName);
 
   DXConStrTabBuilder.finalize();
-  for (auto ElAndName : zip(SignatureElements, SemanticNames)) {
+  for (auto &&ElAndName : zip(SignatureElements, SemanticNames)) {
```

Comment:
The copy here would just be copying a `std::tuple` of iterators (which are pointers). That doesn’t seem like something that should cause any performance issues, and this doesn’t match other idiomatic use cases of `llvm::zip`.